lib/kafka/protocol/record_batch.rb

```diff
@@ -167,8 +167,12 @@ def self.decode(decoder)
         log_append_time = (attributes & TIMESTAMP_TYPE_MASK) != 0
 
         last_offset_delta = record_batch_decoder.int32
-        first_timestamp = Time.at(record_batch_decoder.int64 / 1000)
-        max_timestamp = Time.at(record_batch_decoder.int64 / 1000)
+
+        first_timestamp = record_batch_decoder.int64
+        first_timestamp = Time.at(first_timestamp / 1000, first_timestamp % 1000, :millisecond)
```

Comment:
This syntax is not supported before ruby 2.5. Maybe you should BigDecimal to avoid floating point rounding errors ?

Using something like `seconds/BigDecimal(1000)` in place of just /1000 in all places

Reply:
Thanks, fixed.